src/instrument_meta.py

```diff
@@ -0,0 +1,307 @@
+    if not isinstance(filters, dict):
+        msg = f'Invalid type for filters, must be \'dict\', was ' \
+            f'type: {type(filters)}'
+        raise TypeError(msg)
+
+    print(f'Column \'{key}\' is of type {type(getattr(cls, key).type)}.')
+    int_flt = filters.get(key)
+
+    if int_flt is None:
+        print(f'No \'{key}\' filter detected')
+        return constructed_filter
+
+    constructed_filter[key] = ( getattr(cls, key) == int_flt )
+    
+    return constructed_filter
+
+def construct_filters(filters):
+    constructed_filter = {}
+
+    constructed_filter = get_string_filter(filters, im, 'name', constructed_filter)
+    
+    constructed_filter = get_int_filter(filters, im, 'num_channels', constructed_filter)
+
+    constructed_filter = get_string_filter(filters, im, 'scan_angle', constructed_filter)
+
+    return constructed_filter
+
+
+@dataclass
+class InstrumentMetaRequest:
+    request_dict: dict
+    method: str = field(default_factory=str, init=False)
+    params: dict = field(default_factory=dict, init=False)
+    filters: dict = field(default_factory=dict, init=False)
+    ordering: list = field(default_factory=list, init=False)
+    record_limit: int = field(default_factory=int, init=False)
+    body: dict = field(default_factory=dict, init=False)
+    instrument_meta: InstrumentMeta = field(init=False)
+    response: dict = field(default_factory=dict, init=False)
+
+    def __post_init__(self):
+        self.method = db_utils.validate_method(self.request_dict.get('method'))
+        self.params = self.request_dict.get('params')
+        self.body = self.request_dict.get('body')
+        self.filters = None
+        self.ordering = None
+        self.record_limit = None
+
+        if self.method == db_utils.HTTP_PUT:
+            self.instrument_meta = get_instrument_meta_from_body(self.body)
+        else:
+            if isinstance(self.params, dict):
+                self.filters = construct_filters(self.params.get('filters'))
+                self.ordering = self.params.get('ordering')
+                self.record_limit = self.params.get('record_limit')
+
+                if not type(self.record_limit) == int or self.record_limit <= 0:
+                    self.record_limit = None
+            else:
+                self.filters = None
+                self.ordering = None
+                self.record_limit = None
+            
+
+    def failed_request(self, error_msg):
+        return DbActionResponse(
+            request=self.request_dict,
+            success=False,
+            message='Failed instrument meta request',
+            details=None, 
+            errors=error_msg
+        )
+
+    def submit(self):
+        if self.method == db_utils.HTTP_GET:
+            return self.get_instrument_metas()
+        elif self.method == db_utils.HTTP_PUT:
+            try:
+                return self.put_instrument_meta()
+            except Exception as err:
```

Comment:
specific Exception?

Reply:
not a clear exception, any errors we want to catch and return that the process failed. right now, the code won't crash but will return a failure message (fairly standard practice for an actual API system) if this is something we'd want to change in the future it would need to be a high level architecture decision.